primitives/core/macros/src/lib.rs

```diff
@@ -0,0 +1,77 @@
+#![allow(clippy::tabs_in_doc_comments)]
+
+use proc_macro::TokenStream;
+use reuse::handle_reuse;
+use syn::{parse_macro_input, Error};
+
+mod reuse;
+
+/**
+This macro is used to reuse implementations when the rust's trait system cannot gracefully express the abstraction.
+
+This works similar with `#[cfg(..)]` that sets the target only appear on the specified cases.
+
+# Usage:
+
+```
+use litentry_macros::reuse;
+
+#[reuse(x, y)] . // Define the cases that the following implementation expands for each one
+mod __ {  // Leave mod name with double discards, which is to be replaced by the cases
+	#[x]  // This item would only appear on case `x`
+	fn u() {
+		__
+	}
+
+	#[y]  // This item would only appear on case `y`
+	fn v(a: String) {
+		println!("hello world!")
+	}
+
+	#[x]  // Specifying multiple cases indicates that the item would appear on all of them
+	#[y] .// This behaviour is designed to be different from `#[cfg(..)]`
+	fn a() -> i32 {
+		#[x]  // This statement would only appear on case `x`
+		let p = 1;
+		#[y]  // This statement would only appear on case `y`
+		let p = 2;
+		p + 1
+	}
+
+
+	fn g<#[x] 'a, #[y] T>(#[x] a: i32, #[y] a: u32) {}
+}
+
+```
+Expands to:
+```
+mod x {
+	fn a() -> i32 {
+		let p = 1;
+		p + 1
+	}
+	fn u() {
+		println!("hello world!");
+	}
```

Comment:
@AlseinX would you mind explaining how

```
	#[x]  // This item would only appear on case `x`
	fn u() {
		__
	}
```

expanded to


```
	fn u() {
		println!("hello world!");
	}
```
?

Reply:
As I have commented, leaving a function body with a double underscore placeholder indicates that it reuses the body of the next function.
This make it possible to give a function different signatures by cases.
It is a little bit like typescript's function overloading syntax, that a function may have multiple signatures, but in order to satisfy rust's syntax parser, which does not accept a standalone function signature, there has to be a placeholder function body.

Reply:
Thanks, I probably missed the part that it's the body of the next function.